.github/workflows/avm.platform.publish.tag.yml

```diff
@@ -0,0 +1,63 @@
+name: "avm.platform.publish-tag"
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: "The git tag of the module to publish."
+        required: true
+        type: string
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  job_publish_module_with_tag:
+    runs-on: ubuntu-latest
+    name: "Publish module with tag"
+    steps:
+      - name: Checkout tag
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.tag }}
+
+      - name: Log in to Azure
+        uses: azure/login@v1
+        with:
+          client-id: ${{ env.PUBLISH_CLIENT_ID }}
+          tenant-id: ${{ env.PUBLISH_TENANT_ID }}
+          subscription-id: ${{ env.PUBLISH_SUBSCRIPTION_ID }}
+
+      # Adding a step to explicitly install the latest Bicep CLI because there is
+      # always a delay in updating Bicep CLI in the job runner environments.
+      - name: Install the latest Bicep CLI
+        shell: bash
+        run: |
+          curl -Lo bicep https://github.com/Azure/bicep/releases/latest/download/bicep-linux-x64
+          chmod +x ./bicep
+          sudo mv ./bicep /usr/local/bin/bicep
+          bicep --version
+
+      - name: "Publish tagged module to public bicep registry"
+        uses: azure/powershell@v1
+        with:
+          azPSVersion: "latest"
+          inlineScript: |
+            # Grouping task logs
+            Write-Output '::group::Publish tagged module to public bicep registry'
+
+            # Load used functions
+            . (Join-Path $env:GITHUB_WORKSPACE 'avm' 'utilities' 'pipelines'  'publish' 'Publish-ModuleFromTagToPBR.ps1')
```

Comment:
Wondering about the location of this script. Should we keep it here or move it to the platform folder? Currently modifying the script would trigger all validation pipelines.

Reply:
Well it's a script that is run only by a pipeline for publishing. If we'd move it anywhere else I suspect that it would be rather confusing / hard to find afterwards.

However - we could also exclude it from the trigger / find a smart way to exclude specific scripts? Stupid example: if they'd have 'manual' in the name, they would no trigger runs? 

It's a general challenge. For this script in particular it may not matter much as I doubt it will change much (if at all). But the topic itself is something we could brainstorm.

Reply:
I'm still inclined to have all platform scripts in the platform folder and to not increase trigger complexity. So sure let's discuss offline.

Reply:
Fair enough - to me it's not really worth discussing further. I don't agree, but don't care enough either. I moved it now into a `platform` folder to unblock this PR.